packages/test/test-end-to-end-tests/src/test/orderSequentially.spec.ts

```diff
@@ -0,0 +1,218 @@
+import {
+	ITestObjectProvider,
+	ITestContainerConfig,
+	DataObjectFactoryType,
+	ChannelFactoryRegistry,
+	ITestFluidObject,
+} from "@fluidframework/test-utils";
+import { describeNoCompat } from "@fluid-internal/test-version-utils";
+import { IContainer } from "@fluidframework/container-definitions";
+import { ContainerRuntime } from "@fluidframework/container-runtime";
+import { ConfigTypes, IConfigProviderBase } from "@fluidframework/telemetry-utils";
+import { IValueChanged, SharedDirectory, SharedMap } from "@fluidframework/map";
+import { SharedCell } from "@fluidframework/cell";
+
+const stringId = "sharedStringKey";
+const dirId = "sharedDirKey";
+const cellId = "cellKey";
+const mapId = "mapKey";
+const registry: ChannelFactoryRegistry = [
+	[stringId, SharedString.getFactory()],
+	[dirId, SharedDirectory.getFactory()],
+	[cellId, SharedCell.getFactory()],
+	[mapId, SharedMap.getFactory()],
+];
+const testContainerConfig: ITestContainerConfig = {
+	fluidDataObjectType: DataObjectFactoryType.Test,
+	registry,
+};
+
+describeNoCompat("Multiple DDS orderSequentially", (getTestObjectProvider) => {
+	let provider: ITestObjectProvider;
+	beforeEach(() => {
+		provider = getTestObjectProvider();
+	});
+
+	let container: IContainer;
+	let dataObject: ITestFluidObject;
+	let sharedString: SharedString;
+	let sharedDir: SharedDirectory;
+	let sharedCell: SharedCell;
+	let sharedMap: SharedMap;
+	let changedEventData: IValueChanged[];
+	let containerRuntime: ContainerRuntime;
+	let error: Error | undefined;
+
+	const configProvider = (settings: Record<string, ConfigTypes>): IConfigProviderBase => ({
+		getRawConfig: (name: string): ConfigTypes => settings[name],
+	});
+	const errorMessage = "callback failure";
+
+	beforeEach(async () => {
+		const configWithFeatureGates = {
+			...testContainerConfig,
+			loaderProps: {
+				configProvider: configProvider({
+					"Fluid.ContainerRuntime.EnableRollback": true,
+				}),
+			},
+		};
+		container = await provider.makeTestContainer(configWithFeatureGates);
+		dataObject = await requestFluidObject<ITestFluidObject>(container, "default");
+		sharedString = await dataObject.getSharedObject<SharedString>(stringId);
+		sharedDir = await dataObject.getSharedObject<SharedDirectory>(dirId);
+		sharedCell = await dataObject.getSharedObject<SharedCell>(cellId);
+		sharedMap = await dataObject.getSharedObject<SharedMap>(mapId);
+
+		containerRuntime = dataObject.context.containerRuntime as ContainerRuntime;
+		changedEventData = [];
+		sharedDir.on("valueChanged", (changed, _local, _target) => {
+			changedEventData.push(changed);
+		});
+		sharedCell.on("valueChanged", (value) => {
+			changedEventData.push(value);
+		});
+		sharedCell.on("delete", (value) => {
+			changedEventData.push(value);
+		});
+		sharedMap.on("valueChanged", (value) => {
+			changedEventData.push(value);
+		});
+	});
+
+	it("Should rollback simple edits on multiple DDS types", () => {
```

Comment:
Not a big deal, but just 2-3 is probably sufficient. I don't know that four is usefully testing anything further.